blazar/plugins/floatingips/floatingip_plugin.py

```diff
@@ -447,3 +455,109 @@ def query_fip_allocations(self, fips, detail=None, lease_id=None,
+        return [alloc for alloc
+                in reservation["floatingip_allocations"]
+                if alloc["floating_ip_id"] in resource_ids]
+
+    def get_reservations_by_resource_ids(self, resource_ids,
+                                         interval_begin, interval_end):
+        return db_utils.get_reservations_by_floatingip_ids(resource_ids,
+                                                          interval_begin,
+                                                          interval_end)
+
+    def get_unreservable_resourses(self):
+        return db_api.unreservable_fip_get_all_by_queries([])
+
+    def get_notification_event_types(self):
+        """Get event types of notification messages to handle."""
+        return ['service.update']
+
+    def notification_callback(self, event_type, payload):
+        return {}
+
+    def set_reservable(self, resource, is_reservable):
+        db_api.floatingip_update(resource["id"], {"reservable": is_reservable})
+        LOG.warn(
+            f"{resource['floating_ip_address']} "
+            f"{'recovered' if is_reservable else 'failed'} - setting reservable True"
+        )
+
+    def poll_resource_failures(self):
+        failed = []
+        recovered = []
+        dry_run = CONF[plugin.RESOURCE_TYPE].dry_polling_monitor
+        fips = db_api.floatingip_list()
+
+        def process_fip(fip):
+            fip_address = fip["floating_ip_address"]
+            fip_curr_reservation = db_utils.get_most_recent_reservation_info_by_fip_id(fip['id'])
+            if fip_curr_reservation and fip_curr_reservation['status'] == status.reservation.ACTIVE:
+                # This means that FIP works fine and no need to recover
+                LOG.debug(f"FIP {fip_address} is in active reservation {fip_curr_reservation['id']} - skipping")
+                return
+            fip_pool = neutron.FloatingIPPool(fip['floating_network_id'])
+            # check if the FIP is in neutron subnet allocation pools
+            try:
+                subnet = fip_pool.fetch_subnet(fip_address)
+            # if the FIP is in neutron allocation pools, make the FIP not reservable
+            except utils_exceptions.NeutronUsesFloatingIP as e:
+                LOG.warning(f"Failed: Floating ip {fip_address} is in use by subnet pools")
+                failed.append(fip)
+                return
+            # if the FIP is not found in any subnet, it can be reserved
+            except utils_exceptions.FloatingIPSubnetNotFound as e:
+                LOG.warning(f"Floating ip {fip_address} is not found in any subnet, recovering...")
+                recovered.append(fip)
+                return
+            # get the floating IP reservation ID from neutron
+            try:
+                fip_info_from_neutron = fip_pool.show_floatingip(fip["floating_ip_address"])
+            except Exception as e:
+                # If the FIP is not registered in neutron that means there is no reservation for this IP
```

Comment:
I think this should catch NeutronNotFound or whatever the specific exception is, and the comment should clarify something like:
"If the FIP is not created in neutron, then it is in the correct 'pending' state"
as it's not really about a "no reservation for this IP" but rather the IP is not already in use.